transaction/core/src/blockchain/block_contents.rs

```diff
@@ -0,0 +1,70 @@
+use crate::{blake2b_256::Blake2b256, ring_signature::KeyImage, tx::TxOut, ConvertError};
+use alloc::vec::Vec;
+use core::convert::TryFrom;
+use digestible::{Digest, Digestible};
+use generic_array::{typenum::Unsigned, GenericArray};
+use serde::{Deserialize, Serialize};
+
+/// The contents of a Block.
+#[derive(Clone, PartialEq, Eq, Debug, Digestible, Serialize, Deserialize)]
+pub struct BlockContents {
```

Comment:
Can this derive `Message`? That would simplify things down the line (I think it should just work since TxOut and KeyImage derive it?)